### PR TITLE
safe null support for include and exclude

### PR DESCRIPTION
**Pull request by pwelter34**

### CODE REVIEW

File: src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs [hunk 1]
```diff
@@ -36,7 +36,9 @@ public class FieldIncludesQueryBuilder : IElasticQueryBuilder {
 
     public static class FieldIncludesQueryExtensions {
         public static T IncludeField<T>(this T query, Field field) where T : IFieldIncludesQuery {
-            query.FieldIncludes?.Add(field);
+            if (field != null)
```
Comment (ejsmith) [inline src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs]: Shouldn't we check for null or empty?

File: src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs [hunk 2]
```diff
@@ -48,8 +48,11 @@ public static class FieldIncludesQueryExtensions {
         }
 
         public static T IncludeFields<T>(this T query, params Field[] fields) where T : IFieldIncludesQuery {
-            if (fields != null)
-                query.FieldIncludes?.AddRange(fields);
+            if (fields == null)
+                return query;
+
+            foreach (var field in fields.Where(p => !string.IsNullOrEmpty(p.Name)))
```
Comment (ejsmith) [inline src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs]: Use String.IsNullOrEmpty for consistency

File: src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs [hunk 3]
```diff
@@ -74,8 +77,11 @@ public static class FieldIncludesQueryExtensions {
         }
 
         public static T ExcludeFields<T>(this T query, params Field[] fields) where T : IFieldIncludesQuery {
-            if (fields != null)
-                query.FieldExcludes?.AddRange(fields);
+            if (fields == null)
+                return query;
+
+            foreach (var field in fields.Where(p => !string.IsNullOrEmpty(p.Name)))
```
Comment (ejsmith) [inline src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs]: Use String.IsNullOrEmpty for consistency

Comment (pwelter34) [inline src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs]: is name the correct thing to check?  what if property has a value but not name?

Comment (ejsmith) [inline src/Foundatio.Repositories.Elasticsearch/Queries/Builders/FieldIncludesQueryBuilder.cs]: Yeah, you are right, we can't just check Name. Name isn't populated if you use a lambda expression. So not sure what all you would have to check to see if its empty.

**Pull request merged**